gobblin-kafka/src/main/java/gobblin/kafka/writer/KafkaDataWriter.java

```diff
@@ -38,51 +47,91 @@
 @Slf4j
 public class KafkaDataWriter<D> implements DataWriter<D> {
 
+  private static final long MILLIS_TO_NANOS = 1000 * 1000;
+
   private final KafkaProducer<String, D> producer;
   private final Callback producerCallback;
   private final String topic;
-  private final AtomicInteger recordsWritten;
+  private final long commitTimeoutInNanos;
   private final AtomicInteger bytesWritten;
+  private final AtomicInteger recordsWritten;
+  private long recordsProduced;
```

Comment:
You can use metrics instead (in particular `Counter`). Just extend `InstrumentedDataWriter` and create your custom metrics.